programs/Makefile

```diff
@@ -127,12 +131,15 @@ ifndef WINDOWS
 all: fuzz
 endif
 
-fuzz:
+fuzz: mbedtls_test
 	$(MAKE) -C fuzz THIRDPARTY_INCLUDES=$(THIRDPARTY_INCLUDES)
 
-$(DEP):
+$(MBEDLIBS):
 	$(MAKE) -C ../library
 
+mbedtls_test:
```

Comment:
Question: why introduce this new target of `mbedtls_test` here instead of using `$(MBEDTLS_TEST_OBJS)`, which is what we're linking to? This is inconsistent with what we're doing for the libraries, where we have
```
$(MBEDLIBS):
	$(MAKE) -C ../library
```
so the obvious parallel would be
```
$(MBEDTLS_TEST_OBJS):
	$(MAKE) -C ../tests mbedtls_test
```

Reply:
I made it the same as in tests/Makefile but it is true that the case is different because the objects are not compiled "by this Makefile". Changed to what you were expecting.